nixos/modules/services/x11/hardware/libinput.nix

```diff
@@ -3,23 +3,18 @@
 with lib;
 
 let cfg = config.services.xserver.libinput;
-    xorgBool = v: if v then "on" else "off";
-in {
 
-  options = {
-
-    services.xserver.libinput = {
-
-      enable = mkEnableOption "libinput";
+    xorgBool = v: if v then "on" else "off";
 
+    mkConfigForDevice = deviceType: {
```

Comment:
Not every option makes sense for mouse and touchpad here, so probably we will need some `mkIf deviceType == "touchpad"` and `mkIf deviceType == "mouse"`. But I think this can be fixed in another PR.

Reply:
Yeah, this makes sense. Once this PR is merged I will look at it.